### PR TITLE
feat(cheqd): Add CheqdApi

### PR DESCRIPTION
**Pull request by DaevMithran**
This PR introduces the exposure of the `createResource` function from the `CheqdDidRegistrar` class using `CheqdApi` class, enabling the creation of custom DID-Linked Resources (DLRs). With this enhancement, users can now create and manage DLRs such as trust registries, images, files, and other custom resources as needed.

### CODE REVIEW

File: packages/cheqd/src/CheqdApi.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+@injectable()
+export class CheqdApi {
+  private agentContext: AgentContext
+
+  public constructor(agentContext: AgentContext) {
+    this.agentContext = agentContext
+  }
+
+  public async createResource(did: string, options: CheqdCreateResourceOptions) {
```
Comment (TimoGlastra) [inline packages/cheqd/src/CheqdApi.ts:14]: In the underlying method, all fields except data are optional. Is this correct? Do we need to provide some simplified API top level? OR is it correct that all fields are optiona? (so just providing data should succeed)

Comment (DaevMithran) [reply packages/cheqd/src/CheqdApi.ts:14]: No @TimoGlastra, ideally the agent should provide the below attributes

- name
- resourceType
- data
-  id

They are defaulted to an empty string and will be rejected from the ledger side. Updated `CheqdCreateResourceOptions` with a stricter schema

**Review by TimoGlastra — APPROVED** [review]
Thinking of it now, if we have createResource, shouldn't we also add a getResource method?

**Comment by TimoGlastra** [comment]
I'll merge now, but would  be great if you could also add the getResource method :)

**Comment by TimoGlastra** [comment]
@DaevMithran can you accept my PR to your branch to add a changelog entry? https://github.com/cheqd/credo-ts/pull/6

**Comment by TimoGlastra** [comment]
We can merge after taht

**Pull request merged**